package/boot/uboot-mediatek/patches/454-add-glinet-x3000.patch

```diff
@@ -0,0 +1,290 @@
+diff --git a/arch/arm/dts/mt7981-glinet-gl-x3000.dts b/arch/arm/dts/mt7981-glinet-gl-x3000.dts
+new file mode 100644
+index 0000000..9f35b58
+--- /dev/null
++++ b/arch/arm/dts/mt7981-glinet-gl-x3000.dts
+@@ -0,0 +1,146 @@
++// SPDX-License-Identifier: GPL-2.0
++
++/dts-v1/;
++#include "mt7981.dtsi"
++#include <dt-bindings/gpio/gpio.h>
++#include <dt-bindings/input/linux-event-codes.h>
++
++/ {
++	#address-cells = <1>;
++	#size-cells = <1>;
++	model = "GL.iNet GL-X3000";
++	compatible = "glinet,gl-x3000", "mediatek,mt7981";
++
++	chosen {
++		stdout-path = &uart0;
++		tick-timer = &timer0;
++	};
++
++	memory@40000000 {
++		device_type = "memory";
++		reg = <0x40000000 0x20000000>;
++	};
++
++	reg_3p3v: regulator-3p3v {
++		compatible = "regulator-fixed";
++		regulator-name = "fixed-3.3V";
++		regulator-min-microvolt = <3300000>;
++		regulator-max-microvolt = <3300000>;
++		regulator-boot-on;
++		regulator-always-on;
++	};
++
++	keys {
++		compatible = "gpio-keys";
++
++		reset {
++			label = "reset";
++			linux,code = <KEY_RESTART>;
++			gpios = <&gpio 1 GPIO_ACTIVE_LOW>;
++		};
++	};
++
++	leds {
++		compatible = "gpio-leds";
++
++		wifi2g {
++			label = "green:wifi2g";
++			gpios = <&gpio 30 GPIO_ACTIVE_LOW>;
++		};
++
++		wifi5g {
++			label = "green:wifi5g";
++			gpios = <&gpio 38 GPIO_ACTIVE_LOW>;
++		};
++
++		5g_led1 {
++			label = "green:5g:led1";
++			gpios = <&gpio 6 GPIO_ACTIVE_LOW>;
++		};
++
++		5g_led2 {
++			label = "green:5g:led2";
++			gpios = <&gpio 7 GPIO_ACTIVE_LOW>;
++		};
++
++		5g_led3 {
++			label = "green:5g:led3";
++			gpios = <&gpio 8 GPIO_ACTIVE_LOW>;
++		};
++
++		5g_led4 {
++			label = "green:5g:led4";
++			gpios = <&gpio 4 GPIO_ACTIVE_HIGH>;
++		};
++
++		power {
++			label = "green:power";
++			gpios = <&gpio 39 GPIO_ACTIVE_LOW>;
++		};
++
++		wan {
++			label = "green:wan";
++			gpios = <&gpio 31 GPIO_ACTIVE_LOW>;
++		};
++	};
++};
++
++&eth {
++	status = "okay";
++	mediatek,gmac-id = <1>;
++	phy-mode = "gmii";
++	mediatek,switch = "mt7531";
++	reset-gpios = <&gpio 14 GPIO_ACTIVE_LOW>;
++
++	fixed-link {
++		speed = <1000>;
++		full-duplex;
++	};
++};
```

Comment:
These two devices do not use mt7531.
```
&eth {
        status = "okay";
        mediatek,gmac-id = <1>;
        phy-mode = "gmii";
        phy-handle = <&phy0>;

        mdio {
                phy0: ethernet-phy@0 {
                        compatible = "ethernet-phy-id03a2.9461";
                        reg = <0x0>;
                        phy-mode = "gmii";
                };
        };
};
```

Reply:
Thanks! I've made this change in the device tree and checked tftp booting still works. I'll update the pull request now.